atp/client.go

```diff
@@ -36,13 +36,16 @@ type Client interface {
 	// ReadSchema reads the schema from the ATP server.
 	ReadSchema() (*schema.SchemaSchema, error)
 	// Execute executes a step with a given context and returns the resulting output. Assumes you called ReadSchema first.
-	Execute(input schema.Input, receivedSignals chan schema.Input, emittedSignals chan<- schema.Input) ExecutionResult
+	Execute(input schema.Input, receivedSignals <-chan schema.Input, emittedSignals chan<- schema.Input) ExecutionResult
 	Close() error
 	Encoder() *cbor.Encoder
 	Decoder() *cbor.Decoder
 }
 
 // NewClient creates a new ATP client (part of the engine code).
+// Currently used only by tests in the Python- and Test-deployers.
+//
+//goland:noinspection GoUnusedExportedFunction
 func NewClient(
```

Comment:
And, presumably, this is why only those two deployers were affected by the missing the client-side channel closes?  @jaredoconnell?  🙂